feedback-module/src/components/Module.js

```diff
@@ -84,20 +90,40 @@ function Module({ pagetitle, endpoint, dir }) {
     }
   };
 
+  //Check to see if there are any errors within the form the user tries to submit
+  const formErrors = () => {
+    return (
+      (screen.checkboxes &&
+        screen.checkboxes.required &&
+        !checkboxValidated(checkedFields)) ||
+      !otherFieldValidated() ||
+      (screen.textInputs && !inputsValidated())
+    );
+  };
+
   const submitForm = (nextScreen) => {
+    setOtherTooLong(false);
+    setCheckboxError(false);
     // Submit form data if this screen contains a form
     // Make sure all checkboxes are checked if they exist on this page
-    if (
-      screen.checkboxes &&
-      screen.checkboxes.required &&
-      !checkboxValidated(checkedFields)
-    ) {
-      setCheckboxError(true);
-      firstCheckRef.current && firstCheckRef.current.focus();
-      // Make sure all required fields are completed
-    } else if (screen.textInputs && !inputsValidated()) {
-      focusFirstError();
-    } else {
+    // Make sure all required fields are completed and no inputs are over the character limit
+    if (formErrors()) {
+      if (
+        screen.checkboxes &&
+        screen.checkboxes.required &&
+        !checkboxValidated(checkedFields)
+      ) {
+        setCheckboxError(true);
+        firstCheckRef.current && firstCheckRef.current.focus();
+      } else if (!otherFieldValidated()) {
+        setOtherTooLong(true);
+      }
+      if (screen.textInputs && !inputsValidated()) {
```

Comment:
Is there a reason this is an `if` instead of an `else if`? If there is a missing checkbox and an improper input at the same time, the focus should go to the checkbox since it comes before, but right now it goes to the input instead.

Reply:
If this is an else if then it will only show the checkbox error even if there are two errors

Reply:
I see— should we add an extra check for the checkboxError then, so that we're not accidentally focusing on the input instead of the checkbox? So `if (screen.textInputs && !inputsValidated() && checkboxError)`